### PR TITLE
[dhcp_relay] Update DHCPv6 counter on relayed messages

### PR DESCRIPTION
**Pull request by kellyyeh**
#### Why I did it
DHCPv6 Counter updates counter only on received messages, not relayed messages

#### How I did it
Update counter on messages relayed by DHCPv6 Relay

#### How to verify it
Update counter on messages relayed by DHCPv6 Relay

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [ ] 202106

#### A picture of a cute animal (not mandatory but encouraged)

Signed off by Kelly Yeh: kellyyeh@microsoft.com

### CODE REVIEW

File: src/dhcp6relay/src/relay.cpp [hunk 1]
```diff
@@ -208,19 +208,29 @@ const struct dhcpv6_option *parse_dhcpv6_opt(const uint8_t *buffer, const uint8_
...
  * 
  * @return dhcpv6_option   end of dhcpv6 message option
  */
-void send_udp(int sock, uint8_t *buffer, struct sockaddr_in6 target, uint32_t n) {    
+void send_udp(int sock, uint8_t *buffer, struct sockaddr_in6 target, uint32_t n, relay_config *config, uint8_t msg_type) {
+    std::string counterVlan = counter_table;
     if(sendto(sock, buffer, n, 0, (const struct sockaddr *)&target, sizeof(target)) == -1)
         syslog(LOG_ERR, "sendto: Failed to send to target address\n");
+    else if (msg_type == 0) {
+        syslog(LOG_ERR, "counter: Failed to update correct message counter");
```
Comment (yxieca) [inline src/dhcp6relay/src/relay.cpp:228]: Suggest add msg_type 0 and the title could be 'Unknown'. So you can remove this special handling and tracking how many times unknown  happened.

**Review by yxieca — DISMISSED** [review]

File: src/dhcp6relay/src/relay.cpp [hunk 2]
```diff
@@ -49,7 +49,8 @@ const struct sock_fprog ether_relay_fprog = {
 
 /* DHCPv6 Counter */
 uint64_t counters[DHCPv6_MESSAGE_TYPE_COUNT];
-std::map<int, std::string> counterMap = {{1, "Solicit"},
+std::map<int, std::string> counterMap = {{0, "Unknown"},
```
Comment (yxieca) [inline src/dhcp6relay/src/relay.cpp:52]: Do you need to add initialization code for 'Unknown' in initialize_counter()?

**Review by yxieca — APPROVED** [review]

**Comment by qiluo-msft** [comment]
This commit could not be cleanly cherry-picked to 202012. Please submit another PR.

**Pull request merged**